l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,328 @@
+    ///  and also the necessary approvals are given for the second bridge.
+    /// @notice The logic of this bridge is to allow easy depositing for bridges.
+    /// Each contract that handles the users ERC20 tokens needs approvals from the user, this contract allows
+    /// the user to approve for each token only its respective bridge
+    /// @notice This function is great for contract calls to L2, the secondBridge can be any contract.
+    function requestL2TransactionTwoBridges(
+        L2TransactionRequestTwoBridgesOuter calldata _request
+    ) public payable override nonReentrant returns (bytes32 canonicalTxHash) {
+        {
+            address token = baseToken[_request.chainId];
+
+            if (token == ETH_TOKEN_ADDRESS) {
+                require(msg.value == _request.mintValue + _request.secondBridgeValue, "Bridgehub: msg.value mismatch");
+                // kl todo it would be nice here to be able to deposit weth instead of eth
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken{value: _request.mintValue}(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            } else {
+                require(msg.value == _request.secondBridgeValue, "Bridgehub: msg.value mismatch 2");
+                // note we have to pass token, as a bridge might have multiple tokens.
+                IL1Bridge(baseTokenBridge[_request.chainId]).bridgehubDepositBaseToken(
+                    _request.chainId,
+                    msg.sender,
+                    token,
+                    _request.mintValue
+                );
+            }
+        }
+
+        address stateTransition = getZkSyncStateTransition(_request.chainId);
+        bytes memory data;
+        {
+            bool success;
+            (success, data) = _request.secondBridgeAddress.call{value: _request.secondBridgeValue}(
+                abi.encodeWithSelector(
+                    _request.secondBridgeSelector,
+                    _request.chainId,
+                    msg.sender,
+                    _request.secondBridgeCalldata
+                )
+            );
+            require(success, "Bridgehub: second bridge call failed");
+        }
+        L2TransactionRequestTwoBridgesInner memory outputRequest = abi.decode(
+            data,
+            (L2TransactionRequestTwoBridgesInner)
+        );
+
+        canonicalTxHash = IZkSyncStateTransition(stateTransition).bridgehubRequestL2Transaction(
+            msg.sender,
+            outputRequest.l2Contract,
+            _request.mintValue,
+            _request.l2Value,
+            outputRequest.l2Calldata,
+            _request.l2GasLimit,
+            _request.l2GasPerPubdataByteLimit,
+            outputRequest.factoryDeps,
+            _request.refundRecipient
+        );
+
+        IL1Bridge(_request.secondBridgeAddress).bridgehubConfirmL2Transaction(
```

Comment:
If I provide a malicious `_request.secondBridge`-related data, it can return some invalid `outputRequest.txDataHash`, which can then be used to claim failed deposits

Reply:
But second bridge is called, so second bridge is responsible that two methods are implemented correctly

Reply:
But I can provide any address there, no? I mean, I can deploy my own "malicious bridge" and provide it into that call. I don't see any validation that the second bridge is trusted

Reply:
Ah, nvm. I understood

Reply:
Though it seems like the created L1->L2 transactions are the same regardless of whether the provided address is malicious or not